### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by Subhadip11290157**
Added miriad_compression_codes directory under training extensions/misc/pytorch_toolkit

### CODE REVIEW

File: misc/pytorch_toolkit/miriad_compression_codes/MedPy-0.4.0-py3.8/EGG-INFO/PKG-INFO [hunk 1]
```diff
@@ -0,0 +1,195 @@
+Metadata-Version: 2.1
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/MedPy-0.4.0-py3.8/EGG-INFO/PKG-INFO]: Why is this file here? Isn't this a pip package?

File: misc/pytorch_toolkit/miriad_compression_codes/tests/test_train.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                                                                        self.config['device'],
+                                                                        verbose=True)
+                self.train_bce_list.append(train_loss_bce)
+                self.train_dice_loss_list.append(train_loss_dice)
+                self.train_dice_list.append(train_dice)
+
+            self.assertLessEqual(self.train_bce_list[2], self.train_bce_list[0])
+            self.assertLessEqual(self.train_dice_loss_list[2], self.train_dice_loss_list[0])
+            self.assertGreaterEqual(self.train_dice_list[2], self.train_dice_list[0])
+    return TrainerTest
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/tests/test_train.py]: Add a new line at end of the file.

File: misc/pytorch_toolkit/miriad_compression_codes/tests/test_inference.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            model = inference.load_model(run_type='onnx')
+            mean_dice = inference.inference(model, runtype='onnx')
+            self.assertGreaterEqual(mean_dice, -1)
+
+        def test_ir_inference(self):
+            inference = CustomDatasetPhase1(dataloader_test=self.tst_loader, checkpoint=self.onnx_model_path, device='cpu')
+            model = inference.load_model(run_type='cpu')
+            mean_dice = inference.inference(model, runtype='cpu')
+            self.assertGreaterEqual(mean_dice, -1)
+    return InferenceTest
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/tests/test_inference.py]: Add new line at EOF

File: misc/pytorch_toolkit/miriad_compression_codes/test/test_train.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        else:
+            model = Decoder(int(alpha*depth), int(beta*width))
+
+    return model
+
+
+# validate_model_phase1(config, test_dataloader, model, msecrit)
+
+
+def create_train_test_for_densenet121():
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/test/test_train.py]: densenet121?

File: misc/pytorch_toolkit/miriad_compression_codes/test/test_train.py [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+            config = get_config(action='train')
+            cls.config = config
+            if os.path.exists(config["default_image_path"]):
+                image_path = config["default_image_path"]
+            else:
+                if not os.path.exists(config["image_path"]):
+                    download_data()
+                image_path = config["image_path"]
+
+            dataset_train = RSNADataSet(
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/test/test_train.py]: RSNADataSet?

File: misc/pytorch_toolkit/miriad_compression_codes/test/test_train.py [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+                self.config["max_epoch"], self.config["savepath"])
+            cur_train_loss = self.trainer.current_train_loss
+            self.trainer.train(
+                self.config["max_epoch"], self.config["savepath"])
+            self.assertLessEqual(
+                self.trainer.current_train_loss, cur_train_loss)
+
+    return TrainerTest
+
+
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/test/test_train.py]: Looks like this file was wrongly pushed to PR. If not needed, remove it.

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/train_utils.py [hunk 7]
```diff
@@ -0,0 +1,353 @@
...
+                    colored('[Saving] model saved to {}'.format(model_file), 'red'))
+                torch.save({
+                    'epoch': epoch,
+                    'model_state': model.state_dict(),
+                    'optim_state': optimizer.state_dict(),
+                    'schedular_state': schedular.state_dict(),
+                }, os.path.abspath(model_file))
+                prev_test_ssim = avg_ssim
+            else:
+                print(colored('[Saving] model NOT saved'))
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/src/utils/train_utils.py:370]: Add new line at EOF

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/plot.py [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+	Provide the json file produced by inference.py.
+	(Optionally) Provide name of the plot image. Default is plot.png.
+	"""
+	)
+
+	parse.add_argument('-p', '--plot_json', type=str, required=True, help='the plot json file produced by inference script')
+	parse.add_argument('-o', '--plot_image', type=str, required=False, default='plot.png', help='name of plot image file')
+	args = parse.parse_args()
+
+	main( args )
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/src/utils/plot.py]: new line at EOF

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/model.py [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+            n_downconv=self.n_updownconv, n_encowidth=self.width)
+        self.decoder = Decoder(
+            n_upconv=self.n_updownconv, n_decowidth=self.width)
+
+    def forward(self, x):
+        self.shape_input = list(x.shape)
+        x = self.encoder(x)
+        self.shape_latent = list(x.shape)
+        x = self.decoder(x)
+        return x
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/src/utils/model.py:97]: new line at EOF

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/inference_utils.py [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+        infer_dataset = CustomDatasetPhase2(path_test_latent, path_test_gdtruth, transform_images=images_transforms, transform_masks=labels_transforms, preserve_name=True, mod =  0)
+        infer_dataloader = data.DataLoader(infer_dataset, batch_size=1, num_workers=16, pin_memory=True, shuffle=False)
+
+    model.eval()
+
+    with torch.no_grad():
+		# a global counter & accumulation variables
+        n = 0
+        all_bits, all_bpp, all_ssim, all_psnr = [], [], [], []
+        # avg_bits, avg_bpp, avg_ssim, avg_psnr = 0., 0., 0., 0.
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/src/utils/inference_utils.py]: remove if these commented  variables are not used

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/inference_utils.py [hunk 11]
```diff
@@ -0,0 +1,190 @@
...
+                    json.dump([], json_file)
+
+            with open(json_fillpath, 'r') as json_file:
+                json_content = json.load(json_file)
+
+            # append to the content of json
+            json_content.append({'bits': all_bits, 'bpp': all_bpp, 'ssim': all_ssim, 'psnr': all_psnr})
+
+            with open(json_fillpath, 'w') as json_file:
+                json.dump(json_content, json_file)
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/src/utils/inference_utils.py:219]: New line at EOF

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/export_utils.py [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+
+    def export_model_onnx(self):
+        print(f"Saving model to {self.config.get('model_name_onnx')}")
+        res_path = os.path.join(os.path.split(self.checkpoint)[0], self.config.get('model_name_onnx'))
+        dummy_input = torch.randn(1, 1, 256, 256)
+        torch.onnx.export(self.model, dummy_input, res_path,
+                        opset_version = 11, do_constant_folding=True,
+                        input_names = ['input'], output_names=['output'],
+                        dynamic_axes={'input' : {0 : 'batch_size'}, 'output' : {0 : 'batch_size'}},
+                        verbose = False)
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/src/utils/export_utils.py:39]: new line at EOF

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/evaluators.py [hunk 13]
```diff
@@ -0,0 +1,37 @@
+from skimage.measure import compare_psnr
+import pytorch_msssim, torch
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/src/utils/evaluators.py]: add import torch as new line

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/evaluators.py [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+def compare_ssim_batch( original, compressed, **kwargs ):
+
+	assert original.shape == compressed.shape, 'shapes should be same'
+	assert len(original.shape) == 4 # Batch x Channel x Height x Width
+
+	avg_ssim = pytorch_msssim.msssim(
+		torch.tensor(original, requires_grad=False),
+		torch.tensor(compressed, requires_grad=False), **kwargs).item()
+
+	return avg_ssim
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/src/utils/evaluators.py]: Add new line at EOF

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/download_weights.py [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+
+
+def download_data():
+    config = get_config(action='download')
+    if not os.path.exists('test_data'):
+        os.makedirs('test_data')
+    data_url = config['test_data']['url']
+    data_path = config['test_data']['dest_path']
+    download_and_extract(path=data_path, url=data_url,
+                         expath='test_data/chest_xray_data/')
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/src/utils/download_weights.py]: new line at EOF

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/dataprep.py [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+	)
+
+	parser.add_argument('-p', '--path_to_dataset', type=str, required=True, help='Path to the original CBIS-DDSM folder')
+	parser.add_argument('-d', '--patch_dim', type=int, required=False, default=256, help='Patches of size (patch_dim x patch_dim)')
+	parser.add_argument('--out_train', type=str, required=True, help='Output folder for training samples')
+	parser.add_argument('--out_test', type=str, required=True, help='Output folder for testing samples')
+	parser.add_argument('--report', action='store_true', help='Report statistics')
+
+	args = parser.parse_args()
+	main( args )
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/src/utils/dataprep.py]: Add new line at EOF

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/dataprep.py [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+
+					unint16_count += 1
+					image = image.astype(np.float32) / (2**16 - 1) # normalize b/w [0,1]
+
+					if is_training:
+						# Extract (non-overlapping) patches for training
+						patches = view_as_windows(image, window_shape=(args.patch_dim, args.patch_dim), step=args.patch_dim)
+						for rowid, row in enumerate(patches):
+							for colid, patch in enumerate(row):
+								all_patch_count += 1
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/src/utils/dataprep.py]: Indentation looks a bit weird. Check if it is proper.

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/data_loader.py [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+        if self.transform_images is not None:
+            image = torch.Tensor(image)
+
+        if self.transform_masks is not None:
+            mask = self.transform_masks(mask)
+
+        if self.preserve_name == True:
+            return image, mask, file_join
+        else:
+            return image, mask
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/src/utils/data_loader.py]: New line at EOF

File: misc/pytorch_toolkit/miriad_compression_codes/src/train.py [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+                        required=False, default=1.5, help='Computation power')
+    parser.add_argument('--efficient_net',
+                        action='store_true', help='Run Efficient Net')
+    parser.add_argument('-iter', '--iterate', type=int, required=False,
+                        default=50, help='Grid search iterations for alpha-beta')
+    parser.add_argument('-p', '--phase', type=int,
+                        required=True, default=1, help='Phase 1 or Phase 2')
+    args = parser.parse_args()
+
+    main(args)
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/src/train.py:73]: New line at EOF

File: misc/pytorch_toolkit/miriad_compression_codes/src/inference.py [hunk 20]
```diff
@@ -0,0 +1,71 @@
...
+                        default=3, help='no of conv blocks in the architecture')
+    parser.add_argument('--width', type=int, required=False,
+                        default=64, help='no of Filters')
+    parser.add_argument('-x', '--max_samples', type=int, required=False,
+                        default=0, help='limit the number of samples to use for inference')
+    parser.add_argument('-ph', '--phase', type=int,
+                        required=True, default=1, help='Phase')
+    args = parser.parse_args()
+
+    main(args)
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/src/inference.py:71]: new line at EOF

File: misc/pytorch_toolkit/miriad_compression_codes/requirements.txt [hunk 21]
```diff
@@ -0,0 +1,13 @@
...
+pydicom
+scikit-learn
+tqdm
+pandas
+requests==2.26.0
+openvino-dev[onnx]
+onnxruntime==1.8.1
+numpy==1.19.2
+termcolor
+dataloader
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/requirements.txt]: This seems to be wrong.

File: misc/pytorch_toolkit/miriad_compression_codes/requirements.txt [hunk 22]
```diff
@@ -0,0 +1,13 @@
...
+torchvision==0.6.1
+pydicom
+scikit-learn
+tqdm
+pandas
+requests==2.26.0
+openvino-dev[onnx]
+onnxruntime==1.8.1
+numpy==1.19.2
+termcolor
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/requirements.txt]: What is this?

File: misc/pytorch_toolkit/miriad_compression_codes/model_convert_check.ipynb [hunk 23]
```diff
@@ -0,0 +1,97 @@
+{
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/model_convert_check.ipynb]: What is this ipynb file? Is this needed?

File: misc/pytorch_toolkit/miriad_compression_codes/init_venv.sh [hunk 24]
```diff
@@ -0,0 +1,29 @@
...
+
+cd ${work_dir}
+
+if [[ -e venv ]]; then
+  echo
+  echo "Virtualenv already exists. Use command to start working:"
+  echo "$ . venv/bin/activate"
+fi
+
+virtualenv ${venv_dir} -p python3 --prompt="(chest_x-ray_screening)"
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/init_venv.sh]: Replace chest_xray screening with "compression"

**Review by Rakshith2597 — COMMENTED** [review]
Remove MedPY folder. This is a python package. Why is this folder inside this repo?

Remove model_weights folder before creating pull requests.

Remove pytorch_msssim. use ssim and psnr from this https://torchmetrics.readthedocs.io/en/stable/image/peak_signal_noise_ratio.html

File: misc/pytorch_toolkit/miriad_compression_codes/requirements.txt (hunk 21 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/requirements.txt]: yes, removed it.

File: misc/pytorch_toolkit/miriad_compression_codes/requirements.txt (hunk 22 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/requirements.txt]: "colored" was imported from "termcolor" in src/utils/train_utils.py, and was used for colored tagging of 'training'->yellow, 'saving'->red and 'testing'->cyan

File: misc/pytorch_toolkit/miriad_compression_codes/model_convert_check.ipynb (hunk 23 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/model_convert_check.ipynb]: It was present in the shared drive folder. You are right Sir. Upon reading it, I found that it is exporting the onnx model using torch.onnx.export() which we have already defined in exporter.py and implemented in export.py. So, it seems superfluous. I'll remove it.

File: misc/pytorch_toolkit/miriad_compression_codes/src/inference.py (hunk 20 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/src/inference.py:71]: done, actually when I opened it, I found newline was already present there. But that couldn't be seen in preview mode.

File: misc/pytorch_toolkit/miriad_compression_codes/src/train.py (hunk 19 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/src/train.py:73]: done

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/data_loader.py (hunk 18 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/src/utils/data_loader.py]: done

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/dataprep.py (hunk 16 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/src/utils/dataprep.py]: done

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/download_weights.py (hunk 15 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/src/utils/download_weights.py]: added

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/evaluators.py (hunk 14 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/src/utils/evaluators.py]: added

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/evaluators.py (hunk 13 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/src/utils/evaluators.py]: done

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/export_utils.py (hunk 12 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/src/utils/export_utils.py:39]: done

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/inference_utils.py (hunk 11 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/src/utils/inference_utils.py:219]: done

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/inference_utils.py (hunk 10 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/src/utils/inference_utils.py]: removed as they were not used and weren't affecting the program

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/model.py (hunk 9 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/src/utils/model.py:97]: added

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/plot.py (hunk 8 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/src/utils/plot.py]: added

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/train_utils.py (hunk 7 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/src/utils/train_utils.py:370]: added

File: misc/pytorch_toolkit/miriad_compression_codes/test/test_train.py (hunk 6 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/test/test_train.py]: yes Sir, I deleted it. It was irrelevant for this model.

File: misc/pytorch_toolkit/miriad_compression_codes/test/test_train.py (hunk 5 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/test/test_train.py]: It was my mistake Sir. That was the folder that you advised not to use and it was for a separate task. It has been deleted now and the change will be reflected in the branch.

File: misc/pytorch_toolkit/miriad_compression_codes/tests/test_inference.py (hunk 3 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/tests/test_inference.py]: added

File: misc/pytorch_toolkit/miriad_compression_codes/tests/test_train.py (hunk 2 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/tests/test_train.py]: added

File: misc/pytorch_toolkit/miriad_compression_codes/MedPy-0.4.0-py3.8/EGG-INFO/PKG-INFO (hunk 1 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/MedPy-0.4.0-py3.8/EGG-INFO/PKG-INFO]: I removed the "medpy"-package-related files and folders as they were out of context. 
[ I had included them in the working directory to run a file under the example repo chest_xray_screening before I shifted to the miriad_compression_codes directory. At that time I didn't delete those package-related resources and thus it remained there ]

File: misc/pytorch_toolkit/miriad_compression_codes/init_venv.sh (hunk 24 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/init_venv.sh]: replaced

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/dataprep.py (hunk 17 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/src/utils/dataprep.py]: corrected the indentation

File: misc/pytorch_toolkit/miriad_compression_codes/requirements.txt (hunk 22 above)
Comment (Rakshith2597) [reply misc/pytorch_toolkit/miriad_compression_codes/requirements.txt]: Okay

File: misc/pytorch_toolkit/miriad_compression_codes/configs/phase1_config.json [hunk 25]
```diff
@@ -0,0 +1,127 @@
...
+        ],
+        "model_name_onnx": "miriad.onnx",
+        "model_name": "miriad",
+        "alpha": 1,
+        "beta": 1,
+        "phi": 1,
+        "phase": 1,
+        "class_count": 3
+    }
+}
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/configs/phase1_config.json:73]: Newline

File: misc/pytorch_toolkit/miriad_compression_codes/configs/phase1_config.json [hunk 26]
```diff
@@ -0,0 +1,127 @@
...
+            "0a5b4860-e6ab-4737-a389-9d0cf7300770.jpg": 2,
+            "0a5c4dcb-33ac-4466-9dbf-42ed5c8ec1f0.jpg": 1,
+            "0a6a5956-58cf-4f17-9e39-7e0d17310f67.jpg": 0,
+            "0a7b0cc8-af04-4d2c-9267-6fdfb05f48f2.jpg": 2
+        },
+        "class_names": [
+            "Lung Opacity",
+            "Normal",
+            "No Lung Opacity / Not Normal"
+        ],
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/configs/phase1_config.json]: Remove these lines. Not relevant for this model.

File: misc/pytorch_toolkit/miriad_compression_codes/configs/phase1_config.json [hunk 27]
```diff
@@ -0,0 +1,127 @@
...
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/configs/phase1_config.json]: Remove these lines.

File: misc/pytorch_toolkit/miriad_compression_codes/configs/phase2_config.json [hunk 28]
```diff
@@ -0,0 +1,135 @@
...
+            "0a5b4860-e6ab-4737-a389-9d0cf7300770.jpg": 2,
+            "0a5c4dcb-33ac-4466-9dbf-42ed5c8ec1f0.jpg": 1,
+            "0a6a5956-58cf-4f17-9e39-7e0d17310f67.jpg": 0,
+            "0a7b0cc8-af04-4d2c-9267-6fdfb05f48f2.jpg": 2
+        },
+        "class_names": [
+            "Lung Opacity",
+            "Normal",
+            "No Lung Opacity / Not Normal"
+        ],
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/configs/phase2_config.json]: Remove these lines

File: misc/pytorch_toolkit/miriad_compression_codes/configs/phase2_config.json [hunk 29]
```diff
@@ -0,0 +1,135 @@
...
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/configs/phase2_config.json]: Remove these lines.

File: misc/pytorch_toolkit/miriad_compression_codes/configs/phase2_config.json [hunk 30]
```diff
@@ -0,0 +1,135 @@
...
+        "model_name_onnx": "miriad_eff.onnx",
+        "model_name": "miriad_eff",
+        "alpha": 1,
+        "beta": 1,
+        "phi": 1,
+        "phase": 2,
+        "class_count": 3
+    }
+}
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/configs/phase2_config.json:80]: Add a new line at EOF.

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/dataloader.py [hunk 31]
```diff
@@ -0,0 +1,142 @@
...
+import pandas as pd
+
+
+class CustomDatasetPhase1(data.Dataset):
+	
+    def __init__(self, path_to_dataset, files256=None,files128=None,split=None,
+				transform_images = None, transform_masks = None,
+				images_path_rel = '.', masks_path_rel = '.',
+				preserve_names = False):
+        self.path_to_dataset = os.path.abspath(path_to_dataset) # root folder of the CBIS-DDSM dataset
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/src/utils/dataloader.py]: remove the word CBIS-DDSM from the comments.

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/dataloader.py [hunk 32]
```diff
@@ -0,0 +1,142 @@
...
+    def __getitem__(self, i):
+        # indexing function
+
+        if not hasattr(self, 'cropsubset'):
+            # if not chosen a subset, randomly chose the default 'how_many'
+            self.choose_random_subset()
+
+        # Read the images and masks (same as images)
+        # image = Image.open(os.path.join(self.path_to_dataset, self.images_path_rel, self.cropsubset[i]))
+        # mask = Image.open(os.path.join(self.path_to_dataset, self.masks_path_rel, self.cropsubset[i]))
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/src/utils/dataloader.py]: Remove these commented lines.

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/dataloader.py [hunk 33]
```diff
@@ -0,0 +1,142 @@
...
+            if self.transform_images is not None:
+                image256 = self.transform_images(image256)
+                image128 = self.transform_images(image128)
+            # if self.transform_masks is not None:
+            # 	mask = self.transform_masks(mask)
+
+            # debugging stuff, not important
+            # if self.preserve_names:
+            # 	return image256, image128, self.cropsubset[i]
+            # else:
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/src/utils/dataloader.py]: Remove these lines if they are not used.

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/downloader.py [hunk 34]
```diff
@@ -0,0 +1,36 @@
...
+def download_data(phase):
+    config = get_config(action='download', phase = phase, config_path='configs/')
+    if not os.path.exists('test_data'):
+        os.makedirs('test_data')
+        os.makedirs(os.path.join('test_data', 'prepared'))
+        os.makedirs(os.path.join('test_data', 'prepared', 'bags'))
+    data_url = config['test_data']['drive_url']
+    data_path = config['test_data']['dest_path']
+    download_and_extract(path=data_path, url=data_url,
+                         expath='test_data/rbis_ddsm_sample/')
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/src/utils/downloader.py]: Why rbis_ddsm is mentioned here?

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/get_config.py [hunk 35]
```diff
@@ -0,0 +1,25 @@
...
+        if phase == 1:
+            with open(os.path.join(config_path, 'phase1_config.json')) as f1:
+                config_file = json.load(f1)
+            config = config_file[action]
+        else:
+            with open(os.path.join(config_path, 'phase2_config.json')) as f1:
+                config_file = json.load(f1)
+            config = config_file[action]
+
+    return config
```
Comment (Rakshith2597) [inline misc/pytorch_toolkit/miriad_compression_codes/src/utils/get_config.py]: New line at EOF.

**Review by Rakshith2597 — CHANGES_REQUESTED** [review]
1. Remove the folders phase1 data and phase 2 data. These are downloaded and stored in the respective folders automatically by the downloader.py. Add '.png' in gitignore
2. Remove the folder pytorch_ssim_psnr as they are not being used.

File: misc/pytorch_toolkit/miriad_compression_codes/configs/phase1_config.json (hunk 25 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/configs/phase1_config.json:73]: added newline

File: misc/pytorch_toolkit/miriad_compression_codes/configs/phase1_config.json (hunk 26 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/configs/phase1_config.json]: removed them

File: misc/pytorch_toolkit/miriad_compression_codes/configs/phase1_config.json (hunk 27 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/configs/phase1_config.json]: removed them

File: misc/pytorch_toolkit/miriad_compression_codes/configs/phase2_config.json (hunk 29 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/configs/phase2_config.json]: removed them

File: misc/pytorch_toolkit/miriad_compression_codes/configs/phase2_config.json (hunk 28 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/configs/phase2_config.json]: removed them

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/dataloader.py (hunk 31 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/src/utils/dataloader.py]: removed

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/dataloader.py (hunk 32 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/src/utils/dataloader.py]: removed

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/dataloader.py (hunk 33 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/src/utils/dataloader.py]: removed

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/get_config.py (hunk 35 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/src/utils/get_config.py]: added newline

File: misc/pytorch_toolkit/miriad_compression_codes/src/utils/downloader.py (hunk 34 above)
Comment (Subhadip11290157) [reply misc/pytorch_toolkit/miriad_compression_codes/src/utils/downloader.py]: changed it to expath='test_data/miriad_sample/'
